openerp/addons/base/ir/ir_ui_menu.py

```diff
@@ -329,20 +329,23 @@ def get_needaction_data(self, cr, uid, ids, context=None):
         menu_ids = list(menu_ids)
 
         for menu in self.browse(cr, uid, menu_ids, context=context):
+            enabled = False
+            count = False
             res[menu.id] = {
-                'needaction_enabled': False,
+                'needaction_enabled': enabled,
                 'needaction_counter': False,
             }
             if menu.action and menu.action.type in ('ir.actions.act_window', 'ir.actions.client') and menu.action.res_model:
-                if menu.action.res_model in self.pool:
-                    obj = self.pool[menu.action.res_model]
-                    if obj._needaction:
-                        if menu.action.type == 'ir.actions.act_window':
-                            dom = menu.action.domain and eval(menu.action.domain, {'uid': uid}) or []
-                        else:
-                            dom = eval(menu.action.params_store or '{}', {'uid': uid}).get('domain')
-                        res[menu.id]['needaction_enabled'] = obj._needaction
-                        res[menu.id]['needaction_counter'] = obj._needaction_count(cr, uid, dom, context=context)
+                obj = self.pool.get(menu.action.res_model, None)
+                if obj and obj._needaction:
+                    if menu.action.type == 'ir.actions.act_window':
+                        dom = menu.action.domain and eval(menu.action.domain, {'uid': uid}) or []
+                    else:
+                        dom = eval(menu.action.params_store or '{}', {'uid': uid}).get('domain')
+                    res[menu.id]['needaction_enabled'] = enabled = True
+                    res[menu.id]['needaction_counter'] = count = obj._needaction_count(cr, uid, dom, context=context)
```

Comment:
Why not just set `res[menu.id]` here (with a single literal dict)? That way no need for the temporary variables or for deleting it post-facto.